src/LuaTransformer.ts

```diff
@@ -2165,17 +2165,16 @@ export class LuaTransformer {
             );
         }
 
-        if (ts.isCallExpression(expression) && ts.isPropertyAccessExpression(expression.expression)) {
-            const ownerType = this.checker.getTypeAtLocation(expression.expression.expression);
+        if (
+            ts.isExpressionStatement(statement) &&
```

Comment:
Do you want to disallow it in statement-like places like `() => luaTable.set('x', 'y')`?

Reply:
Could be nice to give it some more features, it there any other statement-like places you can think of?

Reply:
Oh, I have assumed it worked, but it makes sense that it doesn't since it has to return expression, forget about this case then (we'll be able to support it easily with multi-statement expressions, I don't think it worth the complexity for now).

But what I was asking is why you have changed this condition. The way it was written before allowed it to work in cases where `transformExpressionStatement` is called directly with an expression, for example in for statement.

Reply:
It must've been left over from when I was removing the casting from the statements below. This change doesn't seem to disallow LuaTable CallExpressions to be used in for loops and doesn't seem to break anything according to the tests so I'll remove this conditional change